src/protagonist/API/Infrastructure/HydraController.cs

```diff
@@ -77,6 +78,35 @@ protected UrlRoots GetUrlRoots()
             return this.ModifyResultToHttpResult(result, hydraBuilder, instance, errorTitle);
         }, errorTitle);
     }
+
+    /// <summary>
+    /// Handles a deletion
+    /// </summary>
+    /// <param name="request">The request/response to be sent through Mediatr</param>
+    /// <param name="errorTitle">The title of the error</param>
+    /// <param name="cancellationToken">Current cancellation token</param>
+    /// <typeparam name="T">The type of entity that was to be deleted</typeparam>
+    /// ActionResult generated from DeleteResult. This will be 204 on success. Or a Hydra
+    /// error and appropriate status code if failed.
+    protected async Task<IActionResult> HandleDelete<T>(
+        IRequest<(DeleteResult deleteReuslt, string errorMessage)> request,
+        string? errorTitle = "Delete failed",
+        CancellationToken cancellationToken = default)
+        where T : class
+    {
+        return await HandleHydraRequest(async () =>
+        {
+            var result = await Mediator.Send(request, cancellationToken);
+
+            return result.deleteReuslt switch
+            {
+                DeleteResult.NotFound => this.HydraNotFound(),
+                DeleteResult.Error => this.HydraProblem(result.errorMessage, null, 500,
+                    "Delete failed"),
+                _ => NoContent()
```

Comment:
It might be worth having `DeleteResult.Deleted => NoContent()` explicitly and a default handler throw an `ArgumentOutOfRangeException`.

With the above we could add a new enum value (`DeleteResult.Conflict`) and it would return a 204 if the above wasn't updated.